Compiler/types.py

```diff
@@ -5107,6 +5107,104 @@ def update(self, other):
         assert self.m == other.m
         self.v.update(other.v)
 
+
```

Comment:
This class will be removed from types.py as it's not needed. The rabbit comparison logic is added in non_linear.py